File: Bindings/C#/Publisher.cs
using System;
using System.Threading;

   public class Publisher
   {
      public static void Main(string[] args)
      {
         Console.WriteLine("FAMOUSO C# Publisher");
	 famouso.event_t e = new famouso.event_t();
	 e.subject=0xf100000000000000;
	 e.data="P/S-C# Michael";
	 e.len=16;
         Console.WriteLine("Subject 0x{0:X}",e.subject);
	 famouso.PublisherEC pub = new famouso.PublisherEC(e.subject);
	if (pub.announce()>0)
		while(true){
			Console.WriteLine("Publish event 0x{0:X} Length={1} Data={2}",e.subject, e.len, e.data);
	 		e.data="P/S-C# Michael";
			pub.publish(e);
			Thread.Sleep(100);
	 		e.data="P/S-C+ Michael";
			pub.publish(e);
			Thread.Sleep(100);

		}
      }
   }

File: Bindings/C++/Publisher.cc
/*******************************************************************************
 *
 * Copyright (c) 2008-2010 Michael Schulze <mschulze@ivs.cs.uni-magdeburg.de>
 * All rights reserved.
 *
 *    Redistribution and use in source and binary forms, with or without
 *    modification, are permitted provided that the following conditions
 *    are met:
 *
 *    * Redistributions of source code must retain the above copyright
 *      notice, this list of conditions and the following disclaimer.
 *
 *    * Redistributions in binary form must reproduce the above copyright
 *      notice, this list of conditions and the following disclaimer in
 *      the documentation and/or other materials provided with the
 *      distribution.
 *
 *    * Neither the name of the copyright holders nor the names of
 *      contributors may be used to endorse or promote products derived
 *      from this software without specific prior written permission.
 *
 *
 *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
 *    IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
 *    TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
 *    PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
 *    OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
 *    SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
 *    LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
 *    DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
 *    THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 *    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 *    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *
 *
 * $Id$
 *
 ******************************************************************************/

/*
 * Publisher.cc
 *
 * Michael Schulze, 2008
 */

#include "debug.h"
#include <boost/thread/thread.hpp>
#include <boost/thread/xtime.hpp>

// the include contains the famouso
// configuration for local event
// propagation
#include "famouso_bindings.h"

int main(int argc, char **argv) {

    famouso::init<famouso::config>();

    // create a PublisherEventChannel
    // with a specific Subject
    famouso::config::PEC pec(famouso::mw::Subject(0xf100000000000000ull));

    // announce the channel
    pec.announce();

    famouso::mw::Event e(pec.subject());
    e.length = 7;
    e.data = (uint8_t*)"Publish";

    // busy waiting is possible, however giving up
    // the cpu is much more better for other processes
    // that have something to do
    while (1) {
        pec.publish(e);
        boost::xtime time;
        boost::xtime_get(&time, boost::TIME_UTC);
        time.sec += 1;
        boost::thread::sleep(time);
    }
}
